File: src/main/java/com/tronoremunerado/calculator/application/service/BathroomCalculator.java
package com.tronoremunerado.calculator.application.service;

import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Slf4j
@Component
public class BathroomCalculator {
    public double calculateDailyPercentageOfShift(int averageBathroomTime, int numberOfVisitsPerDay, int minutesWorked) {
        log.info("Calculating daily percentage of shift for: averageBathroomTime={}, numberOfVisitsPerDay={}, minutesWorked={}",
                 averageBathroomTime, numberOfVisitsPerDay, minutesWorked);

        double percentage = (averageBathroomTime * numberOfVisitsPerDay * 100) / (double) minutesWorked;
        
        log.info("Daily percentage of shift calculated: {}%", percentage);
        return percentage;
    }
}

File: app/Services/BathroomCalculator.php
<?php

declare(strict_types=1);

namespace App\Services;

final class BathroomCalculator
{
    public function calculateDailyPercentageOfShift(int $averageBathroomTime, int $numberOfVisitsPerDay, int $minutesWorked): float
    {
        if ($averageBathroomTime <= 0 || $numberOfVisitsPerDay <= 0 || $minutesWorked <= 0) {
            return 0.0;
        }

        $percentage = ($averageBathroomTime * $numberOfVisitsPerDay * 100) / $minutesWorked;

        return min(100, $percentage);
    }
}
